### ISSUE TITLE
Consecutive numbering fails

### ISSUE THREAD

**Issue by NeuerDennis** [ether/etherpad-lite#5718]
**Describe the bug**
Instead of getting consecutive numbering in different lines (1. 2. 3.) it stays at "1." (1. 1. 1.).

**To Reproduce**
1. Go to '[...](https://yopad.eu/p)' or a similiar etherpad instance
2. Create a Standard Bulletpoint and write something in the line
3. Press Enter to get to the next line and press Tab once to create a sub-bullet point
4. Now enter the next line without pressing Enter and write something down
5. Mark this line and activate numbering

**Expected behavior**
If you now press enter for the next line the numbering stays at "1." instead of going to "2.". This continues to happen in every other line you create under this.

**Server (please complete the following information):**
 - Etherpad version:
 - OS: [e.g., Ubuntu 20.04]
 - Node.js version (`node --version`):
 - npm version (`npm --version`):
 - Is the server free of plugins: 

**Desktop (please complete the following information):**
 - OS: Windows 10 
 - 19041.1.amd64fre.vb_release.191206-1406 10.0.0.0.2.101
 - Browser Google Chrome, 115.0.5790.32, Google LLC, CHROME.EXE, CHROME.EXE

**Comment by JohnMcLear** [comment]
Confirmed.  Needs test coverage and a fix.

**Comment by arvind9140** [comment]
@NeuerDennis  and @JohnMcLear  I am interested to solve this  issue . Please assign me .

**Comment by SamTV12345** [comment]
> @NeuerDennis and @JohnMcLear I am interested to solve this issue . Please assign me .

I assigned you to the issue.

**Comment by arvind9140** [comment]
I am understand  this issue but  i am not able to find   location of issue file please share issue file location @SamTV12345

**Comment by SamTV12345** [comment]
I'll send the relevant file today or tomorrow. Depends on how long I can stay awake.

**Comment by arvind9140** [comment]
Ok.. I am Waiting.

**Comment by SamTV12345** [comment]
It is done in here: https://github.com/ether/etherpad-lite/blob/f1fea377818ec966a8b6e3c2a11786bb9dfb9133/src/static/js/ace2_inner.js#L67

**Comment by Mahas1234** [comment]
Is the issue still open @SamTV12345  I'm a new contributor can I fix it

**Comment by SamTV12345** [comment]
> Is the issue still open @SamTV12345  I'm a new contributor can I fix it
> 
> 

Yes feel free to fix this :)

**Comment by Suvendu-UI** [comment]
hey @SamTV12345 is this issue still available?

**Comment by SamTV12345** [comment]
Yes

**Comment by siddhant792** [comment]
Is it fixed? I cannot reproduce this using the given steps.

**Comment by Unanemouslyanonymous** [comment]
is this issue open, would like to contribute to this

**Comment by SamTV12345** [comment]
Every issue that is open is still open 😃

**Comment by JohnMcLear** [comment]
> Thanks for clarifying [@JohnMcLear](https://github.com/JohnMcLear). Here's a quick test I whipped up to repro in develop.
> 
> diff --git a/src/tests/frontend-new/specs/ordered_list.spec.ts b/src/tests/frontend-new/specs/ordered_list.spec.ts
> index 04e996e..435a3c7 100644
> --- a/src/tests/frontend-new/specs/ordered_list.spec.ts
> +++ b/src/tests/frontend-new/specs/ordered_list.spec.ts
> @@ -107,3 +107,36 @@ test.describe('ordered_list.js', function () {
>        await expect(padBody.locator('div').first().locator('.list-number2')).toHaveCount(1)
>      });
>    });
> +
> +test('issue 5718 stops numbering OLs after ULs', async function ({page}) {
> +
> +    const padBody = await getPadBody(page);
> +    await clearPadContent(page)
> +
> +    await writeToPad(page, 'Line 1')
> +    await page.keyboard.press('Enter')
> +    await writeToPad(page, 'Line 2')
> +    await page.keyboard.press('Enter')
> +    await writeToPad(page, 'Line 3')
> +
> +    const $insertunorderedlistButton = page.locator('.buttonicon-insertunorderedlist')
> +    const $insertorderedlistButton = page.locator('.buttonicon-insertorderedlist')
> +
> +    const firstLine = padBody.locator('div').nth(7)
> +    const secondLine = padBody.locator('div').nth(8)
> +    const thirdLine = padBody.locator('div').nth(9)
> +
> +    await firstLine.selectText()
> +    await $insertunorderedlistButton.first().click()
> +
> +    await secondLine.selectText()
> +    await $insertorderedlistButton.first().click()
> +
> +    await thirdLine.selectText()
> +    await $insertorderedlistButton.first().click()
> +
> +
> +    expect(await thirdLine.locator('ol').getAttribute('start')).toEqual('3'); // This passes
> +    expect(await thirdLine.locator('ol').getAttribute('class')).toEqual('list-number2'); // This fails
> +
> +});
> 
> First, I don't know why, but `clearPadContent` doesn't seem to do anything for me. I just modified my test to deal with the default content at the top.
> 
> Here's the result when I run it:
> 
> ```
> Error: expect(received).toEqual(expected) // deep equality
> 
> Expected: "list-number2"
> Received: "list-number1"
> 
>   138 |
>   139 |     expect(await thirdLine.locator('ol').getAttribute('start')).toEqual('3');
> > 140 |     expect(await thirdLine.locator('ol').getAttribute('class')).toEqual('list-number2');
>       |                                                                 ^
>   141 |
>   142 | });
>   143 |
>     at etherpad-lite/src/tests/frontend-new/specs/ordered_list.spec.ts:140:65
> ```
> 
> And here's a screenshot of the DOM inspector on those two `ol` elements:
> <img alt="Image" width="532" src="https://private-user-images.githubusercontent.com/1486646/463363780-8939a522-57e3-4822-b2e7-70325a9e1ecd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTIwNTQ5NjUsIm5iZiI6MTc1MjA1NDY2NSwicGF0aCI6Ii8xNDg2NjQ2LzQ2MzM2Mzc4MC04OTM5YTUyMi01N2UzLTQ4MjItYjJlNy03MDMyNWE5ZTFlY2QucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDcwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA3MDlUMDk1MTA1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OGM1ZjY2OWFjOWExZTRlY2ZlNzM4ZTNjNDYxNjFkYzY2OGQyM2Q0M2VmMWYyMWU4YWQ2ZjZmMGYzY2NjYmQ0NCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.9DEAmBU_ZdPr2KfggtlvDd5W3yvTnoXVnmT2NFYZ41Y">
> 
> Notice how the 2nd and 3rd list elements have `start="2"` and `start="3"` respectively, but both have `class="list-number1"`.

Brilliant work, thank you!  @SamTV12345 I think this test can be included with a PR for a fix.  :)

**Comment by egwynn** [comment]
@SamTV12345 the test was designed to catch the error demonstrated in the video. It also has an erroneous assertion at the end that should probably get removed. Does this test fail in your branch in an unexpected way?

**Comment by SamTV12345** [comment]
Yeah it fails at least in the test phase when typing the last text. Do you know where this is located @JohnMcLear ? I'm afk for next three days for business meeting in Berlin, I'll check at the weekend. ☺️ Hopefully we won't break other stuff while fixing the issues but let's see :)

**Comment by egwynn** [comment]
@SamTV12345 as best I can tell, this code is responsible for setting `list-start-numberN`:

https://github.com/ether/etherpad-lite/blob/475456ec10705e7771dd488796a74acb5834411a/src/static/js/domline.ts#L109

**Comment by egwynn** [comment]
@JohnMcLear you might be right. Based on my reading of that code, it seems like it's doing a number of different things depending on who's invoking it and how. It's temping for me (as a relative outsider) to recommend decomposing those responsibilities, but I recognize how daunting such a refactor is likely to be (and how likely it is to introduce regressions given how old the code is). I'm happy to continue to poke at it when I have free time though, and any human-language description of how you remember this code working would be greatly appreciated!
